#!/usr/bin/env python

PACKAGE = "lidar_team_erp42"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()


gen.add("large_st_minPoints", int_t, 0, "large_st_minPoints", 30, 1, 100)
gen.add("large_st_epsilon", double_t, 0, "large_st_epsilon", 0.5, 0, 20)
gen.add("large_st_minClusterSize", int_t, 0, "large_st_minClusterSize", 1000, 1, 100)
gen.add("large_st_maxClusterSize", double_t, 0, "large_st_maxClusterSize", 10000, 1, 100000)


#xMaxROI = 10
gen.add("large_st_xMinROI", double_t, 0, "large_st_xMinROI", 0.5, -200, 10)
gen.add("large_st_xMaxROI", double_t, 0, "large_st_xMaxROI", 9.0, -10, 200)
gen.add("large_st_yMinROI", double_t, 0, "large_st_yMinROI", -1.0, -200, 10)
gen.add("large_st_yMaxROI", double_t, 0, "large_st_yMaxROI", 1.0, -10, 200)
gen.add("large_st_zMinROI", double_t, 0, "large_st_zMinROI", -0.60, -2, 0)
gen.add("large_st_zMaxROI", double_t, 0, "large_st_zMaxROI", 3.0, -2, 100)

gen.add("large_st_xMinBoundingBox", double_t, 0, "large_st_xMinBoundingBox", 0.2, 0, 10);
gen.add("large_st_xMaxBoundingBox", double_t, 0, "large_st_xMaxBoundingBox", 4.0, 0, 10);
gen.add("large_st_yMinBoundingBox", double_t, 0, "large_st_yMinBoundingBox", 0.2, 0, 10);
gen.add("large_st_yMaxBoundingBox", double_t, 0, "large_st_yMaxBoundingBox", 2.5, 0, 10);
gen.add("large_st_zMinBoundingBox", double_t, 0, "large_st_zMinBoundingBox", 0.1, 0, 0.1);
gen.add("large_st_zMaxBoundingBox", double_t, 0, "large_st_zMaxBoundingBox", 2.5, 0, 10);

exit(gen.generate(PACKAGE, "large_static_obstacle", "large_st_hyper_parameter"))
